allennlp/models/semantic_parsing/wikitables/wikitables_semantic_parser.py

```diff
@@ -165,43 +153,16 @@ def forward(self,  # type: ignore
                 target_action_sequences: torch.LongTensor = None) -> Dict[str, torch.Tensor]:
         # pylint: disable=arguments-differ
-        # pylint: disable=unused-argument
-        """
-        In this method we encode the table entities, link them to words in the question, then
-        encode the question. Then we set up the initial state for the decoder, and pass that
-        state off to either a DecoderTrainer, if we're training, or a BeamSearch for inference,
-        if we're not.
-
-        Parameters
-        ----------
-        question : Dict[str, torch.LongTensor]
-           The output of ``TextField.as_array()`` applied on the question ``TextField``. This will
-           be passed through a ``TextFieldEmbedder`` and then through an encoder.
-        table : ``Dict[str, torch.LongTensor]``
-            The output of ``KnowledgeGraphField.as_array()`` applied on the table
-            ``KnowledgeGraphField``.  This output is similar to a ``TextField`` output, where each
-            entity in the table is treated as a "token", and we will use a ``TextFieldEmbedder`` to
-            get embeddings for each entity.
-        world : ``List[WikiTablesWorld]``
-            We use a ``MetadataField`` to get the ``World`` for each input instance.  Because of
-            how ``MetadataField`` works, this gets passed to us as a ``List[WikiTablesWorld]``,
-        actions : ``List[List[ProductionRuleArray]]``
-            A list of all possible actions for each ``World`` in the batch, indexed into a
-            ``ProductionRuleArray`` using a ``ProductionRuleField``.  We will embed all of these
-            and use the embeddings to determine which action to take at each timestep in the
-            decoder.
-        example_lisp_string : ``List[str]``, optional (default=None)
-            The example (lisp-formatted) string corresponding to the given input.  This comes
-            directly from the ``.examples`` file provided with the dataset.  We pass this to SEMPRE
-            when evaluating denotation accuracy; it is otherwise unused.
-        target_action_sequences : torch.Tensor, optional (default=None)
-           A list of possibly valid action sequences, where each action is an index into the list
-           of possible actions.  This tensor has shape ``(batch_size, num_action_sequences,
-           sequence_length)``.
-        """
-
+        raise NotImplementedError
+
+    def _get_initial_state_and_scores(self,
+                                      question: Dict[str, torch.LongTensor],
+                                      table: Dict[str, torch.LongTensor],
+                                      world: List[WikiTablesWorld],
+                                      actions: List[List[ProductionRuleArray]],
+                                      example_lisp_string: List[str] = None,
+                                      add_world_to_initial_state: bool = False) -> Dict:
```

Comment:
Why not just always include the `World`?  It'd make the logic simpler everywhere (e.g., no need for optional inputs to the `State`).

Reply:
I thought not including the world could make the code more efficient. The next PR adds checklist stuff which are all optional fields anyway. I'll get back to this when we clean the code later.